### PR TITLE
Add some more process for breaking changes

### PR DESCRIPTION
**Pull request by abarth**
The additional process adds an explicit approval step for the breaking
nature of the change. The approval can take place on the issue or on the
pull request, whichever is more convenient.

### CODE REVIEW

File: design-principles.md [hunk 1]
```diff
@@ -111,23 +111,40 @@ that improve the situation are allowed to go in.**
...
+    label.
 
-2. A summary of each change.
+ 2. Socialize the idea of making that API break with the project, perhaps by
+    discussing the change in the issue tracker on on the pull request.
 
-3. Clear mechanical steps for porting the code from the old form to the new
-   form, if possible. If not possible, clear steps for figuring out how to port
-   the code.
+ 3. If folks agree that benefits of changing the API outweigh the stablity
```
Comment (tvolkert) [inline design-principles.md]: "the benefits"

**Comment by Hixie** [comment]
The whole label dance seems a bit overly bureaucratic but I agree that we should discuss the changes and get broad approval from interested parties before making API-breaking changes to features that are used.

**Comment by sethladd** [comment]
What do we think about moving this doc to the Wiki? That should make it a bit easier for you to edit.

**Comment by tvolkert** [comment]
> What do we think about moving this doc to the Wiki?

I find the Wiki isn't as discoverable for external contributors.

**Comment by sethladd** [comment]
Easily fixed by making a page called /contributing and linking to all the awesome contributing READMEs and Wiki pages :)

**Comment by abarth** [comment]
> The whole label dance seems a bit overly bureaucratic but I agree that we should discuss the changes and get broad approval from interested parties before making API-breaking changes to features that are used.

Ok, I've changed it to just have one label.

**Comment by sethladd** [comment]
Ready to merge?

**Comment by abarth** [comment]
done

**Pull request merged**